lib/matplotlib/animation.py

```diff
@@ -226,10 +226,14 @@ def saving(self, fig, outfile, dpi, *args, **kwargs):
         '''
         # This particular sequence is what contextlib.contextmanager wants
         self.setup(fig, outfile, dpi, *args, **kwargs)
+        finished = False
```

Comment:
I wonder if `MovieWriter` and friends should be tracking `finished` instead and make cleanup a no-op in that case.